float64 FX
float64 FY
float64 FZ
float64 X
float64 Y
float64 Z
float64 theta